app/presenters/two_factor_auth_code/phone_delivery_presenter.rb

```diff
@@ -95,7 +95,7 @@ def cancel_link
     def troubleshoot_change_phone_or_method_option
       if unconfirmed_phone
         {
-          url: phone_setup_path,
+          url: add_phone_path,
```

Comment:
I think one of the problems we keep running into is that we have separate controllers for setting up a phone for a new account vs. adding one to an existing account. These changes do fix the issue since it allows a user to continue entering a phone, but it still creates a "jump" where someone may start at the `phone_setup` view or `add_phone` view, and here we'll always direct them back to `add_phone`, and not actually where they started from.

This seems like a good fix. I think it also leads nicely into a follow-up to consolidate to one controller, since `Users::PhoneController` could now probably handle the phone setup flow with the change to use `confirm_user_authenticated_for_2fa_setup` instead of `confirm_two_factor_authenticated`? What do you think? Should we write a ticket for that?

Reply:
I agree with consolidating the controllers. Let's write a ticket for that work.